proxy/core/src/main/scala/io/cloudstate/proxy/AkkaActorRegisterFeature.scala

```diff
@@ -5,29 +5,40 @@ import org.graalvm.nativeimage.hosted.Feature.QueryReachabilityAccess
 import org.graalvm.nativeimage.hosted.{Feature, RuntimeReflection}
 
 import scala.collection.JavaConverters._
+import java.lang.reflect.Modifier.isAbstract
 
 @AutomaticFeature
 final class AkkaActorRegisterFeature extends Feature {
+  private[this] final var cache: Set[String] =
+    Set(
+      // FIXME the following are configured in akka-graal-config, remove this when migrating from it
+      "akka.io.TcpConnection",
+      "akka.io.TcpListener",
+      "akka.stream.impl.fusing.ActorGraphInterpreter"
+    )
 
-  override def duringAnalysis(access: Feature.DuringAnalysisAccess) =
-    reachableSubtypes(access, access.findClassByName("akka.akka.Actor")).foreach { cls =>
+  override final def duringAnalysis(access: Feature.DuringAnalysisAccess): Unit = {
+    val akkaActorClass =
+      access.findClassByName(classOf[akka.actor.Actor].getName) // We do this to get compile-time safety of the classes, and allow graalvm to resolve their names
+    if (akkaActorClass != null && access.isReachable(akkaActorClass)) {
       for {
-        context <- getDeclaredField(cls, "context")
-        self <- getDeclaredField(cls, "self")
+        subtype <- access.reachableSubtypes(akkaActorClass).iterator.asScala
+        if subtype != null && !subtype.isInterface && !isAbstract(subtype.getModifiers) && !cache(subtype.getName)
```

Comment:
On second thought I might have been too greedy here, there are cases where abstract actor classes should be added, so that their context/self field can be available to their subclasses?

```suggestion
        if subtype != null && !subtype.isInterface && !cache(subtype.getName)
```